src/features/deposit/components/Form/DepositFormRouter/index.tsx

```diff
@@ -67,70 +95,61 @@ export const DepositFormRouter = ({ onSubmit }: DepositFormRouterProps) => {
       setValue("decimals", data.token.decimals)
       setValue("icon", data.token.icon)
       setValue("symbol", data.token.symbol)
+      onSubmit({
+        ...getValues(),
+      })
+      assetChangeRef.current = false
     }
-  }, [data, setValue])
+  }, [data, setValue, onSubmit, getValues])
+
+  useEffect(() => {
+    const subscription = watch((value, { name }) => {
+      if (name === "blockchain") {
+        setValue("address", "")
+        setValue("decimals", 0)
+        setValue("icon", "")
+        setValue("symbol", "")
```

Comment:
A lot of setValues in a row is a bad sign, options to avoid it:
1. Move to a single useState if data is connected
2. Move to useReducer, so you won't have impossible state
3. Use state library (Zustand, Xstate)